hypervisor/qemu/qmp_wrapper_amd64.go

```diff
@@ -37,53 +35,42 @@ func newNetworkAddSession(ctx *hypervisor.VmContext, qc *QemuContext, id, tapnam
 			Execute: "netdev_add",
 			Arguments: map[string]interface{}{
 				"type":       "vhost-user",
-				"id":         device,
+				"id":         guest.Device,
 				"chardev":    chardevId,
 				"vhostforce": true,
 			},
 		})
-	} else if fd > 0 {
-		scm := syscall.UnixRights(fd)
-		glog.V(1).Infof("send net to qemu at %d", fd)
-		commands = append(commands, &QmpCommand{
-			Execute: "getfd",
-			Arguments: map[string]interface{}{
-				"fdname": "fd" + device,
-			},
-			Scm: scm,
-		}, &QmpCommand{
-			Execute: "netdev_add",
-			Arguments: map[string]interface{}{
-				"type": "tap", "id": device, "fd": "fd" + device,
-			},
-		})
-	} else if tapname != "" {
+	} else {
 		commands = append(commands, &QmpCommand{
 			Execute: "netdev_add",
 			Arguments: map[string]interface{}{
-				"type": "tap", "id": device, "ifname": tapname, "script": "no",
+				"type":   "tap",
+				"script": "no",
+				"id":     guest.Device,
+				"ifname": host.Device,
+				"br":     host.Bridge,
```

Comment:
Is the `br` option a must here? It was added after qemu v2.7. If it's a must, we now require qemu (>2.7).
```
commit 584613eacb3840c1803e665ed7edd4ac186deced
Author: Alexey Kardashevskiy <aik@ozlabs.ru>
Date:   Tue Sep 13 17:11:54 2016 +1000

    tap: Allow specifying a bridge

    The tap backend is already using qemu-bridge-helper to attach tap
    interface to a bridge but (unlike the bridge backend) it always uses
    the default bridge name - br0.

    This adds a "br" property support to the tap backend.

$git describe 584613ea
v2.7.0-679-g584613e
```